--- a/tgn/utils/data_processing.py
+++ b/tgn/utils/data_processing.py
@@ -86,7 +86,7 @@ def get_data(dataset_name, val_ratio, test_ratio, different_new_nodes_between_va
         set(destinations[timestamps > val_time]))
     # Sample nodes which we keep as new nodes (to test inductiveness), so than we have to remove all
     # their edges from training
-    new_test_node_set = set(random.sample(test_node_set, int(0.1 * n_total_unique_nodes)))
+    new_test_node_set = set(random.sample(sorted(test_node_set), int(0.1 * n_total_unique_nodes)))

     # Mask saying for each source and destination whether they are new test nodes
     new_test_source_mask = graph_df.u.map(lambda x: x in new_test_node_set).values
